### PR TITLE
[CSL 880] Quizzes API Support

### PR DESCRIPTION
**Pull request by jjl014**
## Updates:
* Add Quizzes Library Support
* Add/fix tests (All tests pass, except for Recommendations which can probably be ignored cause someone keeps changing the pod for the index that the tests run against 🤦)

## Notes:
* Had to add a quizzes service url (in addition to the normal service url) to the constructorio config object as well as a couple other places in order to support both SABR and quiz requests within a single client instance
* Haven't updated the README yet with quiz examples. Will do so in a separate PR once we're certain that quizzes are a live feature (which I'm not sure if it is considered one yet or not)

### CODE REVIEW

File: library/src/main/java/io/constructor/data/DataManager.kt [hunk 1]
```diff
@@ -185,4 +187,65 @@ constructor(private val constructorApi: ConstructorApi, @ConstructorSdk private
     fun trackRecommendationResultsView(recommendationResultViewRequestBody: RecommendationResultViewRequestBody, params: Array<Pair<String, String>> = arrayOf()): Completable {
         return constructorApi.trackRecommendationResultsView(recommendationResultViewRequestBody, params.toMap())
     }
+
+    fun getNextQuestion(quizId: String, encodedParams: Array<Pair<String, String>> = arrayOf(), preferencesHelper: PreferencesHelper): Observable<ConstructorData<QuizResponse>> {
+        val scheme = preferencesHelper.scheme
+        val serviceUrl = preferencesHelper.quizzesServiceUrl
+        val url = "${scheme}://${serviceUrl}/${ApiPaths.URL_QUIZ_NEXT_QUESTION.format(quizId)}${getAdditionalParamsQueryString(encodedParams)}"
```
Comment (jjl014) [inline library/src/main/java/io/constructor/data/DataManager.kt:194]: This deviates a bit from the normal pattern where we have a dynamic url. In this case, we specify an absolute URL to get around the default service url (`ac.cnstrc.com`) that's used.

File: library/src/main/java/io/constructor/data/model/quiz/QuizResponse.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import java.io.Serializable
+
+/**
+ * Models quiz response
+ */
+@JsonClass(generateAdapter = true)
+data class QuizResponse(
```
Comment (jjl014) [inline library/src/main/java/io/constructor/data/model/quiz/QuizResponse.kt:11]: Break this up into `NextQuestionResponse` and `QuizResultsResponse`

File: library/src/main/java/io/constructor/data/model/quiz/QuizOptionAttribute.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import java.io.Serializable
+
+@JsonClass(generateAdapter = true)
+class QuizOptionAttribute (
+        @Json(name = "name") val name: String?,
+        @Json(name = "value") val value: String?,
+        @Json(name = "images") val images: QuizImages?,
```
Comment (jjl014) [inline library/src/main/java/io/constructor/data/model/quiz/QuizOptionAttribute.kt:11]: Do quiz option attributes actually have images?

Comment (jjl014) [reply library/src/main/java/io/constructor/data/model/quiz/QuizOptionAttribute.kt:11]: Nope, remove!

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 4]
```diff
@@ -605,6 +605,150 @@ object ConstructorIo {
...
+     * ## Example
+     * ```
+     * ConstructorIo.getQuizQuestion("quiz-id", listOf("1", "1,2", "true", "seen", "version-id"))
+     * ```
+     * @param quizId id of the quiz you want to retrieve
+     * @param answers list of answers to send
+     * @param versionId version identifier for the quiz
+     * @param sectionName the section the quiz and results come from. defaults to "Products"
+     */
+    fun getQuizQuestion(quizId: String, answers: List<String>? = null, versionId: String? = null, sectionName: String? = null): Observable<ConstructorData<QuizQuestionResponse>> {
```
Comment (jjl014) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt:618]: `getQuizQuestion` => `getQuizNextQuestion`

**Review by stanlp1 — APPROVED** [review]
Looking good to me. 👍 Would be nice to have 1 more review to check the check 👍

**Review by crgee1 — APPROVED** [review]
lgtm just got some merge conflicts

**Pull request merged**